src/material/dialog/dialog.spec.ts

```diff
@@ -1579,6 +1566,61 @@ describe('MatDialog with default options', () => {
 });
 
 
+describe('MatDialog with animations enabled', () => {
+  let dialog: MatDialog;
+  let overlayContainer: OverlayContainer;
+
+  let testViewContainerRef: ViewContainerRef;
+  let viewContainerFixture: ComponentFixture<ComponentWithChildViewContainer>;
+
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [MatDialogModule, DialogTestModule, BrowserAnimationsModule],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(inject([MatDialog, OverlayContainer], (d: MatDialog, oc: OverlayContainer) => {
+    dialog = d;
+    overlayContainer = oc;
+  }));
+
+  afterEach(() => {
+    overlayContainer.ngOnDestroy();
+  });
+
+  beforeEach(() => {
```

Comment:
Couldn't we combine these 3 different `beforeEach` into one? We could also do it inline since there's only one test in the suite.

Reply:
I took this from other `describe`'s in the dialog test. I simplified this a bit by grouping the last two into a single one. I think we still need at least two to make `inject` work.

Reply:
The `inject` can be called manually too:

```
inject([Provider], (provider: Provider) => {
 ...
})();
```

Reply:
haha, who does this in unit tests? 😄 I'll just stick to what we do in all other dialog tests